apps/protocol-frontend/src/components/WeeklyActiveMembersShell.tsx

```diff
@@ -0,0 +1,64 @@
+import { useContributionList } from '../hooks/useContributionList';
+import { subWeeks, endOfDay, startOfDay } from 'date-fns';
+import { formatDate } from '../utils/date';
+import pluralize from 'pluralize';
+import { TODAY_DATE } from '../utils/constants';
+import CountDisplay from './CountDisplay';
+
+interface WeeklyActiveMembersShellProps {
+  daoId: number;
+}
+
+const WeeklyActiveMembersShell = ({ daoId }: WeeklyActiveMembersShellProps) => {
+  const {
+    data: contributionsInRangeCount,
+    isLoading,
+    isFetching,
+    isError,
+  } = useContributionList({
+    first: 500, //  TODO: want all contributions in range regardless of count -- better way to do this?
+    where: {
+      guilds: { some: { guild: { is: { id: { equals: daoId } } } } },
+      date_of_engagement: { gte: subWeeks(TODAY_DATE, 9) },
+    },
+  });
```

Comment:
I think we should probably handle the aggegation on the backend. It would save fetching all contributions

Reply:
Prisma upports average aggregation https://www.prisma.io/docs/concepts/components/prisma-client/aggregation-grouping-summarizing#aggregate. We need to add a method to calculate the average in protocol client using probably this method https://github.com/Govrn-HQ/govrn-monorepo/blob/staging/apps/protocol-api/src/prisma/generated/type-graphql/resolvers/crud/Contribution/AggregateContributionResolver.ts#L10